test/test-8051-reassembly.sh

```diff
@@ -0,0 +1,20 @@
+BUILD_DIRECTORY=$(mktemp -d tmp.XXXXXXXXXX)
+[ -z "$BUILD_DIRECTORY" ] && exit 1
+mkdir -p $BUILD_DIRECTORY
+
+printf "Reassembling 8051... "
+radare2 -a 8051 -m 0x8000 -e "scr.color=0" -qf -c "pI" ./bins/8051/MP_8192E_FW_NIC.bin > $BUILD_DIRECTORY/MP_8192E_FW_NIC.asm || exit 1
+rasm2 -a 8051 -o 0x8000 -f $BUILD_DIRECTORY/MP_8192E_FW_NIC.asm -B  > $BUILD_DIRECTORY/recompiled-firmware.bin || exit 1
+
+cmp ./bins/8051/MP_8192E_FW_NIC.bin $BUILD_DIRECTORY/recompiled-firmware.bin
+if [ $? -eq 0 ]
+then
+	echo "OK"
+	exit 0
+else
+	radiff2 -q -D ./bins/8051/MP_8192E_FW_NIC.bin "$BUILD_DIRECTORY/recompiled-firmware.bin"
+	echo "failed"
+	exit 1
+fi
+
+rm -rf $BUILD_DIRECTORY
```

Comment:
This line is never executed because you exit before leaving. use RC=0 and then exit $RC